### PR TITLE
App fails to start when it depends on thymeleaf-extras-springsecurity5 but does not have Spring Security on the classpath

### PR DESCRIPTION
**Pull request by terminux**
Closes gh-28947

### CODE REVIEW

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/thymeleaf/ThymeleafAutoConfiguration.java [hunk 1]
```diff
@@ -279,7 +280,7 @@ DataAttributeDialect dialect() {
 	}
 
 	@Configuration(proxyBeanMethods = false)
-	@ConditionalOnClass({ SpringSecurityDialect.class })
+	@ConditionalOnClass({ SpringSecurityDialect.class, AbstractSecurityWebApplicationInitializer.class })
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/thymeleaf/ThymeleafAutoConfiguration.java:283]: I think the intent would be clearer if we checked for one of the classes that `SpringSecurityDialect` depends upon such as `org.springframework.security.web.server.csrf.CsrfToken`.

Comment (terminux) [reply spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/thymeleaf/ThymeleafAutoConfiguration.java]: Done.

**Comment by terminux** [comment]
Thanks @wilkinsona , nice suggestions. I updated the PR and it's ready for your review again.

**Comment by snicoll** [comment]
@terminux thanks for making your first contribution to Spring Boot. There's no need to list all 3 classes so I've only kept what @wilkinsona suggested, see f6f5aec.

**Pull request closed without merging**